src/Google/NotificationsService.php

```diff
@@ -0,0 +1,114 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\Google;
+
+use Automattic\Jetpack\Connection\Client;
+use Automattic\WooCommerce\GoogleListingsAndAds\Infrastructure\Service;
+use Jetpack_Options;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class NotificationsService
+ * This class implements a service to Notify a partner about Shop Data Updates
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\Google\Notifications
+ */
+class NotificationsService implements Service {
+
+	// List of Topics to be used.
+	public const TOPIC_PRODUCT_CREATED   = 'product.created';
+	public const TOPIC_PRODUCT_DELETED   = 'product.deleted';
+	public const TOPIC_PRODUCT_UPDATED   = 'product.updated';
+	public const  TOPIC_COUPON_CREATED   = 'coupon.created';
+	public const  TOPIC_COUPON_DELETED   = 'coupon.deleted';
+	public const  TOPIC_COUPON_UPDATED   = 'coupon.updated';
+	public const  TOPIC_SHIPPING_SAVED   = 'action.woocommerce_after_shipping_zone_object_save';
```

Comment:
This is because of the Google Endpoint specification. They are using https://woocommerce.github.io/woocommerce-rest-api-docs/?shell#webhooks as reference.

`action.woocommerce_after_shipping_zone_object_save` means they are referring to `woocommerce_after_shipping_zone_object_save` action in WC.

Reply:
Maybe there is not a resource named `shipping` (yes they are `product` and `coupon`) or if it exists maybe they didn't know about it. In any case, the expect that one. `action.woocommerce_after_shipping_zone_object_save`